### PR TITLE
A simple implementation of a StorageFactory and a Storage object for in-memory handles.

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: runtime/in-memory-storage.js [hunk 1]
```diff
@@ -12,6 +12,44 @@ import tracing from "../tracelib/trace.js";
 import scheduler from './scheduler.js';
 import util from './recipe/util.js';
 
+class InMemoryKey {
+  constructor(key) {
+    var parts = key.split("://");
+    this.protocol = parts[0];
```
Comment (dstockwell) [inline runtime/in-memory-storage.js]: _protocol?

File: runtime/in-memory-storage.js [hunk 2]
```diff
@@ -12,6 +12,44 @@ import tracing from "../tracelib/trace.js";
...
+  }
+  toString() {
+    return this.protocol + '://' + this.location;
+  }
+}
+
+export class InMemoryStorage {
+  constructor(arc) {
+      this._arc = arc;
+      this.memoryMap = {};
```
Comment (dstockwell) [inline runtime/in-memory-storage.js]: _memoryMap?

File: runtime/in-memory-storage.js (hunk 1 above)
Comment (dstockwell) [reply runtime/in-memory-storage.js]: hmm, do we need to store this here?

File: runtime/manifest.js [hunk 3]
```diff
@@ -32,6 +32,7 @@ class Manifest {
     this._fileName = null;
     this._nextLocalID = 0;
     this._id = null;
+    this.storageProviderFactory = new StorageProviderFactory(this);
```
Comment (dstockwell) [inline runtime/manifest.js]: _

File: runtime/storage-provider-factory.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+// subject to an additional IP rights grant found at
+// http://polymer.github.io/PATENTS.txt
+'use strict';
+
+import {InMemoryStorage} from './in-memory-storage.js';
+
+export default class StorageProviderFactory {
+  constructor(arc) {
+    this._arc = arc;
+    this.storageInstances = {'in-memory': new InMemoryStorage(arc)};
```
Comment (dstockwell) [inline runtime/storage-provider-factory.js]: _

File: runtime/in-memory-storage.js (hunk 1 above)
Comment (shans) [reply runtime/in-memory-storage.js]: This is just a simple data version of the string key.

File: runtime/in-memory-storage.js (hunk 2 above)
Comment (shans) [reply runtime/in-memory-storage.js]: Done.

File: runtime/manifest.js (hunk 3 above)
Comment (shans) [reply runtime/manifest.js]: .

File: runtime/storage-provider-factory.js (hunk 4 above)
Comment (shans) [reply runtime/storage-provider-factory.js]: .

**Pull request merged**